roles/ec2_instance_create_delete/README.md

```diff
@@ -0,0 +1,153 @@
+* ec2:DescribeInstanceAttribute
+* ec2:DescribeInstances
+* ec2:DescribeInstanceStatus
+* ec2:DescribeKeyPairs
+* ec2:DescribeSecurityGroups
+* ec2:DescribeSubnets
+* ec2:DescribeVpcs
+* ec2:DisassociateAddress
+* ec2:ModifyInstanceAttribute
+* ec2:ReleaseAddress
+* ec2:RunInstances
+* ec2:TerminateInstances
+
+## Role Variables
+
+The following variables can be set in the role to customize EC2 instance creation and networking configurations:
+
+* **ec2_instance_create_delete_operation**: (Optional)
+  Target operation for the ec2 instance role. Choices are ["create", "delete"]. Defaults to "create".
+
+* **ec2_instance_create_delete_instance_name**: (Required)
+  The name of the EC2 instance to be created.
+
+* **ec2_instance_create_delete_instance_type**: (Optional)
+  The instance type for the EC2 instance (e.g., `t2.micro`, `m5.large`). Required when `ec2_instance_create_delete_operation` is `create`
+
+* **ec2_instance_create_delete_ami_id**: (Optional)
+  The AMI ID for the EC2 instance. Required when `ec2_instance_create_delete_operation` is `create`
+
+* **ec2_instance_create_delete_key_name**: (Optional)
+  The name of the key pair to use for SSH access to the EC2 instance.
+  If the key does not exist, a key pair will be created with the name.
+  If not provided, instance will not be accessible via SSH.
+  If provided when `ec2_instance_create_delete_operation` is `delete`, the keypair will also be deleted.
+
+* **ec2_instance_create_delete_vpc_subnet_id**: (Optional)
+  The ID of the VPC subnet in which the instance will be launched.
+  If not provided, instance will be created in the default subnet for the default VPC in the AWS region if present.
+
+* **ec2_instance_create_delete_tags**: (Optional)
+  A dictionary of tags to assign to the EC2 instance.
+
+* **ec2_instance_create_delete_wait_for_state**: (Optional)
+  Whether to wait for the EC2 instance to be in the "running" (if creating an instance) or "terminated" (if deleting an instance) state before continuing. Default is `true`.
+
+* **ec2_instance_create_delete_associate_security_groups**: (Optional)
+  List of security group IDs to associate with the EC2 instance.
+
+* **ec2_instance_create_delete_associate_eip**: (Optional)
+  Whether to create an Elastic IP (EIP) and associate it with the EC2 instance. Default is `false`.
+  If true, EC2 instance must be launched in a VPC with an Internet Gateway (IGW) attached, otherwise this will fail. Use [cloud.aws_ops.ec2_networking_resources role](../ec2_networking_resources/README.md) to create the necessary networking resources.
+
+* **ec2_instance_create_delete_eip_tags**: (Optional)
+  Tags to assign to the elastic IP.
+
+## Dependencies
+
+- role: [aws_setup_credentials](../aws_setup_credentials/README.md)
+
+## Examples
+
+Using the role on its own in a playbook:
+
+```yaml
+---
+- name: Create EC2 instance
+  hosts: localhost
+  gather_facts: false
+  roles:
+    - role: cloud.aws_ops.ec2_instance_create
```

Comment:
```suggestion
    - role: cloud.aws_ops.ec2_instance_create_delete
```

Reply:
I +1 for `manage_ec2_instance` and was thinking the same.

Reply:
Yep, I totally agree. Will update the role name.

Reply:
What do we think about `ec2_instance_provision`? The role doesn't really provide full instance management...